frontend/src/containers/Subscription/SubscriptionFeed.js

```diff
@@ -28,102 +28,112 @@ class SubscriptionFeed extends Component {
     componentDidMount() {
         this.props.onGetSubscriptions()
             .then(() => {
-                this.setState({
-                    subscriptions: this.props.subscriptionItems,
-                });
+                this.setState((prevState) => ({
+                    subscriptions: prevState.subscriptions.concat(this.props.subscriptionItems),
+                    recoCount: 30 - this.props.subscriptionItems.length,
+                }));
                 this.props.onGetRecommendations()
                     .then(() => {
-                        this.setState({
-                            recommendations: this.props.recommendationItems,
+                        this.setState((prevState) => ({
+                            recommendations: prevState.recommendations.concat(
+                                this.props.recommendationItems,
+                            ),
+                        }), () => {
```

Comment:
[Question] Are they executed in parallel or sequentially?

Reply:
About api calls? They are executed sequentially.

Reply:
I mean, about two `then`s.

Reply:
It is more close to sequential access. They do not work in parallel and the second then would start after onGetRecommendations in the first then passes Promise(I don't know how exactly this should be called)

Reply:
@kosumin OK, but if we cannot assure the first `then` is executed before the second, it can be problems because `addRecoToSub()` has dependency with the first. How about moving this part
```

this.setState((prevState) => ({
                             recommendations: this.props.recommendationItems,		                             recommendations: prevState.recommendations.concat(
                                 this.props.recommendationItems,
                             ),
                         })
```
into `addRecoToSub()`? I think this part always precedes the function in this container.

Reply:
And maybe it is safer to assign the concatenated recommendations to a local variable, then use it for `setState()` and instructions of `addRecoToSub()`. I'm not sure, but I think changing states sometimes aren't reflected immediately.

Reply:
We can assure that the first then is executed before the second because anyway the second is in the first and addToReco happens after setState in the second then finishes because I used the pattern setState callback. So we don't have to worry about that situation! https://reactjs.org/docs/react-component.html#setstate

Reply:
Aha, I didn't know about the setState callback pattern. Thank you for letting me know!